R/install_sys_deps.R

```diff
@@ -0,0 +1,186 @@
+    miniSh = glue('Miniconda3-latest-{miniOsSh}-x86_64.sh')
+    withr::local_file(miniSh)
+    download.file(
+      glue('https://repo.anaconda.com/miniconda/{miniSh}'), miniSh, quiet = TRUE)
+    system2('sh', c(miniSh, '-b', '-p', file.path(installDir, 'miniconda3')))
+
+    cat('Running conda init...\n')
+    system(glue('{minicondaPath}/bin/conda init bash'))}
+
+  # Create new environment if it doesn't exist
+  if (minicondaEnv != 'base' && !dir.exists(minicondaEnvPath)) {
+    cat('Creating conda environment...\n')
+    yamlPath = system.file('extdata', 'conda_env.yml', package = 'seeker')
+    envYaml = yaml::read_yaml(yamlPath)
+
+    if (minicondaEnv != envYaml$name) {
+      yamlPath = withr::local_tempfile(fileext = '.yml')
+      envYaml$name = minicondaEnv
+      yaml::write_yaml(envYaml, yamlPath)}
+
+    system(glue('{minicondaPath}/bin/conda env create -f "{yamlPath}"'))}
+
+  # Set the option
+  if (setSeekerOption) {
+    addToProfile(
+      glue('options(seeker.miniconda = "{minicondaEnvPath}")'), type = 'R')
+    options(seeker.miniconda = minicondaEnvPath)}
+
+  cat('Installing conda packages via mamba...\n')
+  mambaEnvPath = system.file('extdata', 'mamba_env.yml', package = 'seeker')
+  mambaArgs = c('env', 'update', '-p', minicondaEnvPath, '--file', mambaEnvPath)
+  system3('mamba', mambaArgs)
+
+  invisible()}
+
+
+setRefgenie = function(refgenieDir) {
+  cat('Configuring refgenie...\n')
+  if (!dir.exists(refgenieDir)) dir.create(refgenieDir)
+  refgenieYamlPath = file.path(path.expand(refgenieDir), 'genome_config.yaml')
+  if (!file.exists(refgenieYamlPath)) {
+    system3('refgenie', c('init', '-c', refgenieYamlPath))}
+  addToProfile(glue('export REFGENIE="{refgenieYamlPath}"'))
+  addToProfile(glue('Sys.setenv(REFGENIE = "{refgenieYamlPath}")'), type = 'R')
+  Sys.setenv(REFGENIE = refgenieYamlPath)
+  invisible()}
+
+
+getSalmonIndexes = function(salmonIndexes) {
+  cat('Fetching salmon indexes...\n')
+  for (salmonIndex in salmonIndexes) {
+    rgArgs = c('pull', salmonIndex, '--genome-config', Sys.getenv('REFGENIE'))
+    system3('refgenie', rgArgs)}
+  invisible()}
+
+
+# sraToolkitPath - checks path and then installs.
+# minicondaPath - checks path and then installs
+# install as environment called seeker or have as argument.
+# minicondaEnv = "seeker" - installs and uses seeker environment by default,
+# if "base" doesn't create new environment and uses base env.
+# If specified, installs all mamba packages by default
+# Add option setting to ~/.Rprofile so it sets every time.
+# refgenieDir - directory to initialize refgenie to. Checks first then runs init.
+# salmonIndexes - indexes to pull using refgenie, requires regenieDir to be set.
+# fastqscreenDir - Dir to pass to -outdir
+installSysDeps = function(
+    sraToolkitDir = '~', minicondaDir = '~', minicondaEnv = 'seeker',
+    refgenieDir = '~/genomes', salmonIndexes = NULL, fastqscreenDir = NULL) {
```

Comment:
It would be simpler, but I don't want to make assumptions about where to download such a huge amount of data, particularly since people might be running seeker on servers or clusters.

I did just slightly tweak the logic for refgenie, though.